src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,226 @@
+                        String time = parsed[4];
+                        String[] startEndTime = time.split("-");
+                        String startTime = startEndTime[0];
+                        LocalTime localStartTime = LocalTime.parse(startTime);
+
+                        if (startEndTime.length < 2) {
+                            if (doneIndicator.equals("1")) {
+                                task = new Event(taskName, true, localDate, localStartTime);
+                            } else {
+                                task = new Event(taskName, false, localDate, localStartTime);
+                            }
+                        } else {
+                            String endTime = startEndTime[1];
+                            LocalTime localEndTime = LocalTime.parse(endTime);
+                            if (doneIndicator.equals("1")) {
+                                task = new Event(taskName,
+                                    true, localDate, localStartTime, localEndTime);
+                            } else {
+                                task = new Event(taskName,
+                                    false, localDate, localStartTime, localEndTime);
+                            }
+                        }
+                    }
+
+                } else if (identifier.equals("D")) {
+                    //String dateAndTime = parsed[3];
+                    String date = parsed[3];
+                    //String[] dateTime = dateAndTime.split(" ");
+                    //String date = dateTime[0];
+                    LocalDate localDate = LocalDate.parse(date);
+
+                    if (parsed.length < 5) {
+                        if (doneIndicator.equals("1")) {
+                            task = new Deadline(taskName, true, localDate);
+                        } else {
+                            task = new Deadline(taskName, localDate);
+                        }
+                    } else {
+                        String time = parsed[4];
+                        LocalTime localTime = LocalTime.parse(time);
+                        if (doneIndicator.equals("1")) {
+                            task = new Deadline(taskName, true, localDate, localTime);
+                        } else {
+                            task = new Deadline(taskName, false, localDate, localTime);
+                        }
+                    }
+
+                } else {
+                    throw new StorageException("Invalid format. Moving on to the next task.");
+                }
+
+            }
+            return task;
+
+        } catch (DateTimeParseException e) {
+            throw new CalendarException("Please input the correct date and time format. "
+                + "YYYY-MM-DD for date and HH:MM for time.");
+        }
+
+    }
+
+    /**
+     * Updates the data in the file.
+     * @param tasks List containing all the updated tasks.
+     */
+    public static void updateData(List<Task> tasks) {
```

Comment:
Perhaps you could shorten this method by extracting some LOC into several other methods or classes?